### PR TITLE
[FEAT] Upgrade Island

### PR DESCRIPTION
**Pull request by adamhannigan**
# Description

This PR includes the logic to upgrade your island - `farm.upgraded`

Screenshot 2024-01-17 at 10 49 08 am
Screenshot 2024-01-17 at 10 49 11 am

**Excludes**

- Exact requirements
- Expanding on new island + resources

## How to test?

1. Go to Grubnuk on the raft
2. Confirm Upgrade

### CODE REVIEW

**Review by c-py — CHANGES_REQUESTED** [review]
Having a hard time testing - getting stuck when upgrading, the trees are crashing my game. Do you know what causes this?

treecrash

File: src/features/game/expansion/components/IslandUpgrader.tsx [hunk 1]
```diff
@@ -0,0 +1,281 @@
...
+            transition: "opacity 1.25s ease-in-out",
+          }}
+          className={classNames(
+            "bg-black absolute z-10 inset-0  opacity-0 pointer-events-none flex justify-center items-center",
+            {
+              "opacity-100": showTravelAnimation,
+            }
+          )}
+        >
+          <span className="loading">Exploring</span>
```
Comment (c-py) [inline src/features/game/expansion/components/IslandUpgrader.tsx:222]: Some inspiration:
firelighter-export

Comment (adamhannigan) [reply src/features/game/expansion/components/IslandUpgrader.tsx:222]: So elegant!

File: src/features/game/events/landExpansion/upgradeFarm.ts [hunk 2]
```diff
@@ -0,0 +1,451 @@
...
+export function upgrade({ state, action, createdAt = Date.now() }: Options) {
+  let game = cloneDeep(state) as GameState;
+
+  const upcoming = ISLAND_UPGRADE[game.island.type];
+
+  if (game.inventory["Basic Land"]?.lt(upcoming.expansions)) {
+    throw new Error("Player has not met the expansion requirements");
+  }
+
+  // Check & burnthe requirements
```
Comment (brian-man) [inline src/features/game/events/landExpansion/upgradeFarm.ts:393]: Minor: "burn the"

**Pull request merged**